app/src/common/shared/com/igalia/wolvic/downloads/DownloadsManager.java

```diff
@@ -134,6 +146,69 @@ public void startDownload(@NonNull DownloadJob job) {
         }
     }
 
+    public void downloadBlobUri(DownloadJob job) {
+        if (!UrlUtils.isBlobUri(job.getUri()) || job.getInputStream() == null) {
```

Comment:
You can get rid of the first check since you're already checking that before calling the method. I think we should consider that a precondition of the method.